dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -396,7 +393,7 @@ Spdp::shutdown()
       }
       purge_handshake_deadlines(part);
 #endif
-      remove_discovered_participant(part++);
+      purge_discovered_participant(part);
     }
   }
 
```

Comment:
Does it make sense to clear `participants_` here since purge doesn't do an erase like remove did?

Reply:
Since we are in `shutdown`, I don't think it is necessary.  However, if it makes the code more boring than I'm for it.